docs/modules/sql/pages/sql-reflection-configuration.adoc

```diff
@@ -0,0 +1,96 @@
+= Configuring Reflection
+:description: Clusters can be configured to restrict the capability of creating objects through reflection during SQL execution.
```

Comment:
we should be clear here that this applies only to `format=java`. Eg. zero-config compact serialization reflection will not be affected (secured)

Reply:
dc36063bc06516632afd7329dbc04854ec1e3174

Reply:
it is unclear that this applies to SQL mappings and UDTs. IMO writing only `format=java` is too generic.

Reply:
a376deb8e8a6aefeef679dd5796b743893931703

Reply:
https://github.com/hazelcast/hz-docs/pull/955#discussion_r1453461783 ?